File: 8kyu/(8 kyu) Century From Year/(8 kyu) Century From Year.java
// 1 Plain solution
public class Solution {
  public static int century(int number) {
    return number % 100 == 0
      ? number / 100
      : (number / 100) + 1;
  }
}

// 2 Optimized solution
public class Solution {
  public static int century(int year) {
    return (year + 99) / 100;
  }
}

// 3 Clever solution
public class Solution {
  public static int century(int number) {
    return (int) Math.ceil((double) number / 100);
  }
}

File: 8kyu/(8 kyu) Century From Year/(8 kyu) Century From Year.dart
// 1 Plain solution
int century(year) {
  final century = year % 100 == 0 ? year ~/ 100 : year ~/ 100 + 1;
  return century;
}

// 2 Optimized solution
int century(int year) {
  return (year + 99) ~/ 100;
}

// 3 Clever solution
int century(int year) {
  return (year / 100).ceil();
}

// 4 Coding golf
int century(int y) => (y + 99) ~/ 100;
